### DISCUSSION TITLE
Demo Database - Script error

### DISCUSSION THREAD

**Discussion by GST-00** [team-moeller/better-access-charts#63] [Q&A] [answered]
Hi, 

(Ref: version 1.32.05)

I found this resource today and looks like it might help me with a project I have been considering - so firstly thank you for putting this together. I used to use access but got frustrated many years ago with the charting limitations, and switched over to Excel (presentation is important). This week I hit the 'integer' data type limit in excel for the first time (which I then solved) but it put Access back in the front of mind, so I thought I would look into it. This looks like it could be a great tool. 

Anyway, I downloaded the demo and unpacked it. 
Opened "Better Access Charts.accdb", clicked "Create Chart" and got: 
"An error has occurred in the script on this page. Line2, Char: 5, Error: Object doesn't support property or method 'getContext', Code 0. " 

It seems to be this line in the script: 
    var ctx = document.getElementById('myChart').getContext('2d');

(there are 2 other subsequent errors, but these may be caused by the first one). 

I also tried  changing the script source references to the local files, but it did not change the result. 
I'm not sure if there is something else I need to do or install  in the set-up to make the demo work, or is it a foible with my particular version of office, or system configuration? It's probably something absurdly simple, but I can't figure out what it is. 

I'm sure you'll want to know about my system..
Win10 Pro,  10.0.19043, i7 x64, Microsoft 365 Family Subscription, Microsoft® Access® for Microsoft 365 MSO (Version 2110 Build 16.0.14527.20234) 64-bit 
Default browser: Firefox 93.0 (64-bit)

**Comment by team-moeller** [comment, marked as answer]
Hi,

the error sound familiar to me. You will get this error because of the webbrowser control. The webbrowser control emulates the internet explorer IE 7. Here java script is not fully supported. Normally this should be not a problem because there is a line in the html file that ensures that IE=Edge is used. Here java script is fully supported.

On my own computer I get this error from time to time. I have installed Access 2010 parallel to Office 365. The solution is to set a value in the registry.

Daniel Pineault describes this here:
https://www.devhut.net/everything-you-never-wanted-to-know-about-the-access-webbrowser-control/#Emulation

This works well for me. So, I am confident it will be a valid solution for you too.

Best Regards

Thomas

**Reply by GST-00** [reply]
Hi,

That registry edit did the trick, so thanks to you and to Daniel, and especially appreciative of how quickly you replied. 
Could I suggest this be added to the Pre-requisites page at:
 <https://github.com/team-moeller/better-access-charts/wiki/Prerequisites-to-get-started>.
Regards,
G

**Reply by team-moeller** [reply]
I followed your hint: Today I have revised the "Feature Browser Emulation" section on the [Prerequisites page](https://github.com/team-moeller/better-access-charts/wiki/Prerequisites-to-get-started).

**Reply by KentGorrell** [reply]
I had the same error, 
The Access roadmap has a new browser control sheduled so that should resolve this issue
but until then, as I cannot ensure that a registry edit can be performed on my client's devices, instead I decided to output to an html file (same as your option for Save to html file for debugging) and bound the control source of the browser control to that. This works just fine. No error.

to implement this, I added a public property to the class to return the full path of the html file to use when refreshing the browser control in the form code
```
Public Property Get HTML_FullPath() As String
Const HtmlFileName As String = "BetterAccessChart_{}.html" ' let's move make this to make it a module level const
     HTML_FullPath = CurrentProject.Path & "\" & Replace(HtmlFileName, "{}", m_Control.Name)
End Property
```

the requery method of the control does not refresh from the file, it is only meant to requery from data changes, so we need to set the controlsource to something else and then reset it to the full path when the file is recreated.

having just struggled with making "Modern Charts" work in Access of late, I gotta say that Better Access Charts IS BRILLIANT!.
a big thank you

**Reply by team-moeller** [reply]
Hi Kent!
Today I published a new release where the problem with ist fixed. Please have a look at version 1.33.13.
Best Regards and a happy new year!
Thomas

**Reply by KentGorrell** [reply]
Thanks Tom,
I do like the use of 
`Control.Object.Navigate`

**Reply by KDGundermann** [reply]
Hi Thomas,
should we do the registry hack from code when BAC is used for the first time ?
 I assume only Office 2010 and older is effected?? 
Could you please give us your Windows/Office Versions when you encounter this error ?
